metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1262,6 +1260,55 @@ class MetalsLanguageServer(
             val msg = s"Invalid arguments: $args. Expecting: $argExample"
             Future.failed(new IllegalArgumentException(msg)).asJavaObject
         }
+      case ServerCommands.NewScalaWorksheet() =>
+        val args = params.getArguments.asScala
+        (args match {
+          case Seq(JsonString.Option(directory), JsonString(name)) =>
+            val result =
+              newFilesProvider.createWorksheet(directory.map(new URI(_)), name)
+            result.onFailure {
+              case NonFatal(e) =>
+                languageClient
+                  .showMessage(
+                    MessageType.Error,
+                    s"Cannot create worksheet:\n ${e.toString()}"
+                  )
+            }
+            result
+          case _ =>
+            val msg = s"Invalid arguments: $args."
+            Future.failed(new IllegalArgumentException(msg))
+        }).asJavaObject
+      case ServerCommands.NewScalaClass() =>
```

Comment:
Is it possible to może most of the logic to the new provider you created?

Reply:
Te code grew a bit and we need to reduce the amount of code in MetalsLanguageServer I think.